src/IB/IBFEMethod.cpp

```diff
@@ -354,24 +343,50 @@ build_ib_ghosted_system_data(std::vector<SystemData>& ghosted_system_data,
     return;
 }
 
+auto insert_parallel_into_ghosted = [](const PetscVector<Number>& parallel_vector,
+                                       PetscVector<Number>& ghosted_vector) {
+    TBOX_ASSERT(parallel_vector.size() == ghosted_vector.size());
+    TBOX_ASSERT(parallel_vector.local_size() == ghosted_vector.local_size());
+    ghosted_vector = parallel_vector;
+    ghosted_vector.close();
+};
+
+auto convert_parallel_to_ghosted = [](const std::string& vector_name, System& system) {
+    std::unique_ptr<NumericVector<double> > clone_vector;
+    bool from_restart = RestartManager::getManager()->isFromRestart();
+    if (from_restart)
+    {
+        NumericVector<double>* current = system.request_vector(vector_name);
+        if (current != nullptr)
+        {
+            clone_vector = current->clone();
+        }
+    }
+    system.remove_vector(vector_name);
+    system.add_vector(vector_name, /*projections*/ true, /*type*/ GHOSTED);
+
+    if (clone_vector != nullptr)
+    {
+        const auto& parallel_vector = dynamic_cast<const PetscVector<Number>&>(*clone_vector);
+        auto& ghosted_vector = dynamic_cast<PetscVector<Number>&>(system.get_vector(vector_name));
+        insert_parallel_into_ghosted(parallel_vector, ghosted_vector);
+    }
+};
```

Comment:
AFAICT we shouldn't need to use these in IBFEMethod any more. If we do then we should promote them to protected static functions inside `FEMechanicsBase` so that we don't need to duplicate code.

Reply:
These calls are needed to setup all of the vectors used in the different systems. I've renamed the key function `setup_system_vectors` and moved it into the base class as a static member function.